docs/azure/sdk/azure-sdk-thread-safety.md

```diff
@@ -0,0 +1,124 @@
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddControllersWithViews();
+
+    var blobServiceClient = new BlobServiceClient(new Uri("<secrets_endpoint>"), new DefaultAzureCredential());                   
+    services.AddSingleton(blobServiceClient);
+          
+    // Additional code ...
+}
+
+```
+
+For more information about implementing dependency injection with the Azure SDK, refer to the article [Dependency injection with the Azure .NET SDK](./dependency-injection.md).
+
+Alternatively, you may also create the instance of an SDK client and pass it around as a parameter to methods in need of a client. The point is, avoid unnecessary instantiations of the same SDK client object with the same parameters as it's unnecessary and wasteful.
+
+## Clients are not disposable
+
+Two final questions that often come up are:
+
+* Do I need to dispose of Azure SDK client objects when I'm finished using them?
+* Why aren't HTTP-based Azure SDK client objects disposable?
+
+Internally, all Azure SDK clients use a single shared `HttpClient` instance. The clients don't create any other resources that need to be actively freed. The shared `HttpClient` instance persists throughout the entire application lifetime.
+
+```csharp
+// Both clients reuse the shared HttpClient and don't need to be disposed
+var blobClient = new BlobClient(new Uri(sasUri));
+var blobClient2 = new BlobClient(new Uri(sasUri2));
+```
+
+It is possible to provide a custom instance of `HttpClient` to an Azure SDK client object.  In this case, then you become responsible for managing the `HttpClient` lifetime and properly disposing of it at the right time.  
+
+```csharp
+var httpClient = new HttpClient();
+
+var clientOptions = new BlobClientOptions()
+{
+    Transport = new HttpClientTransport(httpClient)
+}
+
+// Both client would use the HttpClient instance provided in clientOptions
+var blobClient = new BlobClient(new Uri(sasUri), clientOptions);
+var blobClient2 = new BlobClient(new Uri(sasUri2), clientOptions);
+
+// ...
+
+// you are responsible for properly disposing httpClient some time later
+httpClient.Dispose();
+```
+
+Further guidance for properly managing and disposing of `HttpClient` instances can be found in the <xref:System.Net.Http.HttpClient> documentation.
+
+## See also
+
+- [Lifetime management for Azure SDK .NET clients (Azure SDK blog)](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/)
+- [Best practices for using Azure SDK with ASP.NET Core (Azure SDK blog)](https://devblogs.microsoft.com/azure-sdk/best-practices-for-using-azure-sdk-with-asp-net-core/)
```

Comment:
Our blog posts typically capture the state of things at a point in time. Because they aren't updated, I suggest removing these blog post links. The content will be updated in docs.

```suggestion
```